src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -331,47 +331,56 @@ internal static List<CompletionResult> MakeCommandsUnique(IEnumerable<PSObject>
             List<CompletionResult> endResults = null;
             foreach (var keyValuePair in commandTable)
             {
-                var commandList = keyValuePair.Value as List<object>;
-                if (commandList != null)
+                if (keyValuePair.Value is List<object> commandList)
                 {
-                    endResults ??= new List<CompletionResult>();
-
-                    // The first command might be an un-prefixed commandInfo that we get by importing a module with the -Prefix parameter,
-                    // in that case, we should add the module name qualification because if the module is not in the module path, calling
-                    // 'Get-Foo' directly doesn't work
-                    string completionName = keyValuePair.Key;
-                    if (!includeModulePrefix)
+                    var modulesWithCommand = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                    var importedModules = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                    var commandInfoArray = new CommandInfo[commandList.Count];
+                    for (int i = 0; i < commandInfoArray.Length; i++)
```

Comment:
I think it's better use `commandList.Count` for this loop
```suggestion
                    for (int i = 0; i < commandList.Count; i++)
```